src/identity_tree/status.rs

```diff
@@ -0,0 +1,156 @@
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+/// The status pertains to the status of the root.
+/// But it can also be used interchangeably with the status of an identity
+/// as all identity commitments have an associated root.
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub enum Status {
+    /// Root is included in sequencer's in-memory tree, but is not yet
+    /// mined.
+    Pending,
+
+    /// Root is mined on mainnet but is still waiting for confirmation on
+    /// relayed chains
+    ///
+    /// i.e. the root is included in a mined block on mainnet,
+    /// but the state has not yet been bridged to Optimism and Polygon
+    ///
+    /// NOTE: If the sequencer is not configured with any secondary chains this
+    /// status should immediately become Finalized
+    Processed,
+
+    /// Root is mined and relayed to secondary chains
+    Mined,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub enum UnprocessedStatus {
+    /// Unprocessed identity failed to be inserted into the tree for some reason
+    ///
+    /// Usually accompanied by an appropriate error message
+    Failed,
+
+    /// Root is unprocessed - i.e. not included in sequencer's
+    /// in-memory tree.
+    New,
+}
+
+/// A status type visible on the API level - contains both the processed and
+/// unprocessed statuses
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum ApiStatus {
+    Unprocessed(UnprocessedStatus),
+    Processed(Status),
+}
```

Comment:
If this pertains to the API/server, what do you think about moving it into the server module rather than in the identity tree?

Reply:
It's used by the api but not only - it's just a way to unify these 2 types but also keep them separate where it's important

Reply:
I suppose we could make it so that
`ApiStatus -> Status`
`Status -> ProcessedStatus` but I'm not sure if I'm a huge fan of that

Reply:
Actually I changed it to just that